src/Init/Data/List/BasicAux.lean

```diff
@@ -12,60 +12,138 @@ namespace List
 /-! The following functions can't be defined at `Init.Data.List.Basic`, because they depend on `Init.Util`,
    and `Init.Util` depends on `Init.Data.List.Basic`. -/
 
-def get! [Inhabited α] : List α → Nat → α
+/--
+Returns the `i`-th element in the list (zero-based).
+
+If the index is out of bounds (`i ≥ as.length`), this function panics when executed, and returns
+`default`. See `get?` and `getD` for safer alternatives.
+-/
+def get! [Inhabited α] : (as : List α) → (i : Nat) → α
   | a::_,  0   => a
   | _::as, n+1 => get! as n
   | _,     _   => panic! "invalid index"
 
-def get? : List α → Nat → Option α
+/--
+Returns the `i`-th element in the list (zero-based).
+
+If the index is out of bounds (`i ≥ as.length`), this function returns `none`.
+Also see `get`, `getD` and `get!`.
+-/
+def get? : (as : List α) → (i : Nat) → Option α
   | a::_,  0   => some a
   | _::as, n+1 => get? as n
   | _,     _   => none
 
-def getD (as : List α) (idx : Nat) (a₀ : α) : α :=
-  (as.get? idx).getD a₀
+/--
+Returns the `i`-th element in the list (zero-based).
+
+If the index is out of bounds (`i ≥ as.length`), this function returns `a₀`.
+Also see `get?` and `get!`.
+-/
+def getD (as : List α) (i : Nat) (a₀ : α) : α :=
+  (as.get? i).getD a₀
+
+/--
+Returns the first element in the list.
 
+If the the list is empty, this function panics when executed, and returns `default`.
+See `head` and `headD` for safer alternatives.
```

Comment:
If the list is empty, this function panics when executed, crashing in compiled programs
and returning `default` from the perspective of proofs.

Reply:
```suggestion
If the list is empty, this function panics when executed, crashing in compiled programs
and returning `default` from the perspective of proofs.
```